amalgamate.py

```diff
@@ -154,8 +160,15 @@ def merge_sources(*, source_dir: Path, covered_headers: Set[Path]):
 
     return output
 
+def do_polyfill(content):
+    return content.replace('<expected>', '<expected.hpp>') \
```

Comment:
I recommend keeping it as ``<tl/expected.hpp>`` because users can fetch the header file from package managers or Fetch-Content in cmake. You can add an option to download the file from the repo so it won't have an impact on those users, the rest looks good to me

Reply:
I think it is cleaner without another directory. This file isn't going to be regularly updated. It hasn't had any major changes since 2 years ago and in a few years, I would expect C++23 to be widespread enough to drop this feature.

Reply:
How about compromising and supporting both via `_has_include`? See https://github.com/cursey/safetyhook/blob/629558c64009a7291ba6ed5cfb49187086a27a47/src/inline_hook.cpp#L3-L9